src/ducks/remoteConfig.ts

```diff
@@ -1,26 +1,62 @@
 import { logger } from "config/logger";
 import { isAndroid } from "helpers/device";
-import { getBundleId } from "react-native-device-info";
+import { getBundleId, getVersion } from "react-native-device-info";
 import { ANALYTICS_CONFIG } from "services/analytics/constants";
 import { getExperimentClient } from "services/analytics/core";
 import { create } from "zustand";
 
 const ONE_HOUR_IN_MS = 60 * 60 * 1000;
 
-interface RemoteConfigState {
-  // Feature flags
+// Boolean feature flags type
+type BooleanFeatureFlags = {
   swap_enabled: boolean;
   discover_enabled: boolean;
   onramp_enabled: boolean;
+};
+
+// String feature flags type
+type StringFeatureFlags = {
+  required_app_version: string;
+  latest_app_version: string;
+};
+
+// Complex feature flags type
+type ComplexFeatureFlags = {
+  app_update_text: {
+    enabled: boolean;
+    payload: string | undefined;
+  };
+};
+
+// Combined feature flags type
+type FeatureFlags = BooleanFeatureFlags &
+  StringFeatureFlags &
+  ComplexFeatureFlags;
+
+interface RemoteConfigState extends FeatureFlags {
+  // State
+  isInitialized: boolean;
   // Actions
   fetchFeatureFlags: () => Promise<void>;
   initFetchFeatureFlagsPoll: () => void;
+  setInitialized: (initialized: boolean) => void;
 }
 
-type FeatureFlags = Omit<
-  RemoteConfigState,
-  "fetchFeatureFlags" | "initFetchFeatureFlagsPoll"
->;
+// Get current app version for default values
+const currentAppVersion = getVersion();
+
+// Boolean feature flags configuration
+const BOOLEAN_FLAGS = [
+  "swap_enabled",
+  "discover_enabled",
+  "onramp_enabled",
+] as const;
+
+// String feature flags configuration (simple string values)
+const STRING_FLAGS = ["required_app_version", "latest_app_version"] as const;
+
+// Complex feature flags configuration (objects, JSON, etc.)
+const COMPLEX_FLAGS = ["app_update_text"] as const;
```

Comment:
would there be a way to automate those array creations based on the `BooleanFeatureFlags`, `StringFeatureFlags` and `ComplexFeatureFlags` keys? So devs don't need to remember to add new flags to both places